### DISCUSSION TITLE
How to install actions ( formerly called the nightly)

### DISCUSSION THREAD

**Discussion by jantje** [Sloeber/arduino-eclipse-plugin#1663] [Show and tell]
UPDATE: You have to be logged in on github to download the nightly.

You may have heard things like "fixed in the nightly" and tried the nightly site and noticed the nightly site still is on version 4.

The fact is: I should call it actions instead of nightly but ... I claim guilty for being: stubborn.
The reason the nightly is no longer updated is that my build server has been down since 7 September 2022 and I stopped trying to revive it. 
In the mean time @wimjongman implemented github actions (thanks Wim) which made github the build server.

The builds on github are now the "nightly" builds. (I'll try to use actions but no guarantees).
You can click on actions at the top of all sloeber github pages and get the latest successful master build (or pull request branch that can be merged in Master).
![image](https://github.com/user-attachments/assets/f597245d-21f2-4064-ac23-f1812935f996)
Click on a successful build (there is currently a issue that only exists on the build server I'm investigating)
Now you can download the repository (you need this to use the eclipse update method) or the sloeber product for you OS to start from fresh.

![image](https://github.com/user-attachments/assets/89432574-e17e-4c4d-9544-1ffc93bbbdcb)
To use the repository: after download unzip the file. In eclipse add the repository in the preferences to the repositories or select install new software; point to the unzipped repository and eclipse should say this becomes a update.

If there are any issues with this feel free to add comments below.
Jantje

Note that due to my build server being down baeyens.it only gets updated on a manual basis. Typically that is at release time.
Note that if you download a product version from baeyens.it it actually comes from github.
You can find them here
![image](https://github.com/user-attachments/assets/47aa5067-4e85-4d07-96a6-88e14e17ab61)

**Comment by arthurfprecht** [comment]
Hi Jantje! Thanks a lot for the guide. Maybe it would be good to add a little detail: To be able to download the nightly packages (Github calls them artifacts but I don't like the naming either, too abstract for me), the user **must be logged in**. Otherwise, the packages are listed but the download button does not appear.  All the best!

**Reply by jantje** [reply]
That is true.
At the time of writing I was not aware of that.
Thanks for pointing this  out.
